src/tools/pkcs11-tool.c

```diff
@@ -1946,6 +1946,42 @@ static void verify_signature(CK_SLOT_ID slot, CK_SESSION_HANDLE session,
 
 	close(fd2);
 
+	if (opt_mechanism == CKM_ECDSA || opt_mechanism == CKM_ECDSA_SHA1 ||
+		opt_mechanism == CKM_ECDSA_SHA256 || opt_mechanism == CKM_ECDSA_SHA384 ||
+		opt_mechanism == CKM_ECDSA_SHA512 || opt_mechanism == CKM_ECDSA_SHA224) {
+		if (opt_sig_format && (!strcmp(opt_sig_format, "openssl") ||
+							   !strcmp(opt_sig_format, "sequence"))) {
+
+			CK_BYTE* bytes;
+			CK_ULONG len;
+			size_t rs_len = 0;
+			unsigned char rs_buffer[512];
+			bytes = getEC_POINT(session, key, &len);
+			free(bytes);
+			/*
+			 * (We only support uncompressed for now)
+			 * Uncompressed EC_POINT is DER OCTET STRING of "04||x||y"
+			 * So a "256" bit key has x and y of 32 bytes each
+			 * something like: "04 41 04||x||y"
+			 * Do simple size calculation based on DER encoding
+			 */
+			if ((len - 2) <= 127)
+				rs_len = len - 3;
+			else if ((len - 3) <= 255)
+				rs_len = len - 4;
+			else
+				util_fatal("Key not supported");
```

Comment:
isn't this length check performed by `sc_asn1_sig_value_sequence_to_rs`? I think that's the natural place to put it.

Reply:
It seems that all the callers (card-isoApplet.c and card-myeid.c) to sc_asn1_sig_value_sequence_to_rs() have the same difficulty to determine the length (buflen) to give the function.
card-isoApplet.c is doing some calculation based on ec_field_length, card-myeid.c is decoding the ASN1...
It would be simpler if we do a first call to the function with a NULL buf, allocate the buffer and then call it a second time if the correct length.